src/certificates/layout/header-buttons/hooks/useHeaderButtons.jsx

```diff
@@ -0,0 +1,48 @@
+import { useState, useMemo } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+
+import {
+  getCourseModes, getCertificateActivationUrl, getCertificateWebViewUrl, getIsCertificateActive,
+} from '../../../data/selectors';
+import { updateCertificateActiveStatus } from '../../../data/thunks';
+
+const useHeaderButtons = () => {
+  const { courseId } = useParams();
+  const dispatch = useDispatch();
+  const courseModes = useSelector(getCourseModes);
+  const certificateWebViewUrl = useSelector(getCertificateWebViewUrl);
+  const certificateActivationHandlerUrl = useSelector(getCertificateActivationUrl);
+  const isCertificateActive = useSelector(getIsCertificateActive);
+
+  const [dropdowmItem, setDropdowmItem] = useState(courseModes[0]);
+
+  const handleActivationStatus = () => {
+    const status = !isCertificateActive;
+
+    dispatch(updateCertificateActiveStatus(courseId, certificateActivationHandlerUrl, status));
+  };
+
+  const previewUrl = useMemo(() => {
+    if (!certificateWebViewUrl) { return ''; }
+
+    const url = new URL(certificateWebViewUrl, window.location.origin);
```

Comment:
```suggestion
    const url = () => new URL(certificateWebViewUrl, window.location.origin);
```

Reply:
There have been previous issues where URL was not created when it was not part of a function. See #706 for details about how it might throw an error

Reply:
Thanks, i see, i will fix it now